src/Analyzers/CSharp/Analyzers/UsePatternCombinators/AnalyzedPattern.cs

```diff
@@ -92,13 +92,7 @@ private Binary(AnalyzedPattern leftPattern, AnalyzedPattern rightPattern, bool i
                 if (!SyntaxFactory.AreEquivalent(target.Syntax, rightPattern.Target.Syntax))
                     return null;
 
-                // We factor out not-patterns in binary combinators.
-                // For instance: `not a and not b` is simplified as `not (a or b)`.
-                // Likewise, `not a or not b` is simplified as `not (a and b)`.
-
-                return (leftPattern, rightPattern) is (Not left, Not right)
-                    ? Not.Create(new Binary(left.Pattern, right.Pattern, !isDisjunctive, token, target))
-                    : new Binary(leftPattern, rightPattern, isDisjunctive, token, target);
+                return new Binary(leftPattern, rightPattern, isDisjunctive, token, target);
```

Comment:
i think it's a worth a comment that we are explicitly not trying to simplify here as user may not find the change substantively better for their domain.

Reply:
when we do want to do that, it'll be in another pass, not the factory method.